core/src/partition_cfg.rs

```diff
@@ -12,33 +12,49 @@ use std::sync::RwLock;
 pub struct Partition {
     pub num_partitions: usize,
     pub my_partition: usize,
-    pub start_ts: u64,
-    pub end_ts: u64,
+    pub partition_start_epoch_length: u64,
+    pub partition_duration_ms: u64,
+    start_ts: u64,
     leaders: Arc<RwLock<Vec<Pubkey>>>,
+    pub done: bool,
 }
 impl Default for Partition {
     fn default() -> Self {
         Self {
             num_partitions: 0,
             my_partition: 0,
             start_ts: 0,
-            end_ts: 0,
             leaders: Arc::new(RwLock::new(vec![])),
+            partition_duration_ms: 0,
+            partition_start_epoch_length: 0,
+            done: false,
         }
     }
 }
 
 #[derive(Default, Debug, Clone)]
 pub struct PartitionCfg {
     partitions: Vec<Partition>,
+    printed: HashSet<u64>,
 }
 
 impl PartitionCfg {
     pub fn new(partitions: Vec<Partition>) -> Self {
-        Self { partitions }
+        Self {
+            partitions,
+            printed: HashSet::new(),
+        }
+    }
+    pub fn all_partitions_done(&self) -> bool {
```

Comment:
looks like this can be removed

Reply:
this whole file could be removed I think.